onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -124,6 +125,25 @@ TEST(GraphTransformationTests, ConstantFolding1) {
   ASSERT_TRUE(op_to_count["Unsqueeze"] == 0);
 }
 
+TEST(GraphTransformationTests, ShapeToInitializer1) {
+  string model_uri = MODEL_FOLDER + "shape-abs-id.onnx";
```

Comment:
Would be good to have a test where the input to Shape had a symbolic dim to make sure things don't break.

Reply:
Added a new graph with three Shapes. One of them has an input with symbolic dimension, so it's not being eliminated.